### PR TITLE
[17.01] Fix order of PATH when doing hacky Conda activating.

### PR DESCRIPTION
**Pull request by jmchilton**
Thanks to @yhoogstrate's comment here https://github.com/galaxyproject/galaxy/commit/fe2a1d8ade4fbda158ae035c78133258d7fec099#commitcomment-20900607.

### CODE REVIEW

File: lib/galaxy/tools/deps/resolvers/conda.py [hunk 1]
```diff
@@ -325,7 +325,7 @@ def shell_commands(self, requirement):
             # On explicit testing the only such requirement I am aware of is samtools - and it seems to work
             # fine with just appending the PATH as done below. Other tools may require additional
```
Comment (nsoranzo) [inline lib/galaxy/tools/deps/resolvers/conda.py]: s/appending/prepending/

Comment (jmchilton) [reply lib/galaxy/tools/deps/resolvers/conda.py]: Rebased with this fix, thanks!

**Comment by mvdbeek** [comment]
I'm not crazy about this change. IUC shouldn't be putting samtools in the global PATH in such a hacky way as we do currently. If a python binary happens to be in the samtools environment (admittedly that shouldn't happen), we're back to square one. I think the proper fix is to install samtools into __samtools@uv in the travis recipe.

**Comment by jmchilton** [comment]
Regardless of how the IUC should configure its testing - don't you think binaries from "required" Conda recipes should take precedent over generic background ones that might not even be setup for Galaxy - let alone that particular requirement?

**Comment by mvdbeek** [comment]
But this only happens for tools that are marked to preserve the python environment. So in fact this already happens?!

**Review by nsoranzo — APPROVED** [review]

**Comment by mvdbeek** [comment]
See also https://github.com/galaxyproject/galaxy/commit/fe2a1d8ade4fbda158ae035c78133258d7fec099#commitcomment-20931952.
I think my first comment is slightly besides the point. I think the whole preserve python environment thing is there to make an exception and keep python off the PATH. So i think it should come last.

**Comment by mvdbeek** [comment]
So the tool in question is SAM-to-BAM, which happens to be on the preserve python list. But the current version doesn't need an importable galaxy ... I'm OK to merge this, and then we can augment the mechanism to take into account the version as well.

**Comment by jmchilton** [comment]
> I think the whole preserve python environment thing is there to make an exception and keep python off the PATH. So i think it should come last.

That actually makes sense to me - now I see.

The better fix here is to apply that hack only to older versions of the tool - that logic needs to be added anyway.

I'm leaning away from merging this PR.

**Pull request closed without merging**